service/routers/users.js

```diff
@@ -0,0 +1,74 @@
+import { Router } from 'express'
+import { sanitize } from 'express-validator'
+import { User } from '../models/User.js'
+
+const usersRouter = Router()
+
+function sanitizeUsers(users) {
+  const sanitizedUsers = users.map((user) => ({
+    id: user._id,
+    firstName: user.firstName,
+    lastName: user.lastName,
+    email: user.email,
+    decks: user.decks,
+    active: user.active
+  }))
+  return sanitizedUsers;
+} 
+
+const getUsers = async (req, res) => {
+  const { userId } = req.user
+  const requestor = await User.findById(userId)
+  if (requestor.role === 'admin' || requestor.role === 'user') {
+    const users = await User.find({})
+    res.send(sanitizeUsers(users))
+  } else {
+    res.status(403).send('Forbidden')
+  }
+}
+
+const getUsersById = async (req, res) => {
+  const { userId } = req.user
+  const requestor = await User.findById(userId)
+  if (requestor.role === 'admin' || requestor.role === 'user') {
```

Comment:
What use case are you missing?

Reply:
I am not sure what use case I am missing, all 'user' actions should be done through updateUser()

Reply:
The implication of the role descriptions in the assignment is that an admin or superuser should be able to access user data for any user but a user should only be able to see their own data. Hence (requestor.role === 'admin' || requestor.role === 'superuser' || (requestor.role === 'user' && requestor._id === req.params.id))